src/nupic/algorithms/Connections.cpp

```diff
@@ -442,23 +444,26 @@ void Connections::computeActivity(
 void Connections::adaptSegment(const Segment segment, 
                                const SDR &inputs,
                                const Permanence increment,
-                               const Permanence decrement)
+                               const Permanence decrement,
+                               const vector<Permanence> &previousUpdates,
+                                     vector<Permanence> &currentUpdates)
 {
-  const vector<Synapse> &synapses = synapsesForSegment(segment);
-
   const auto &inputArray = inputs.getDense();
 
-  for (SynapseIdx i = 0; i < synapses.size(); i++) {
-    const SynapseData &synapseData = dataForSynapse(synapses[i]);
+  for( const auto &synapse : synapsesForSegment(segment) ) {
+    const SynapseData &synapseData = dataForSynapse(synapse);
 
-    Permanence permanence = synapseData.permanence;
+    Permanence update;
     if( inputArray[synapseData.presynapticCell] ) {
-      permanence += increment;
+      update = increment;
```

Comment:
For your consideration: efficiency vs. readability (vs what the compiler does?), please test this in godbolt.org ASM : 
you could assign Permanence update = decrement; 
and only use 1 if: update = -decrement;

If there's no binary-code difference, the current is better as it's clearer what the intention is.

Reply:
I trust my compiler to know how to deal with `if else` statements.  I'm keeping what's here because it is more readable.